src/character.cpp

```diff
@@ -730,6 +730,24 @@ bool Character::has_martialart( const matype_id &m ) const
     return martial_arts_data->has_martialart( m );
 }
 
+int Character::count_threshold_substitute_traits() const
+{
+    int count = 0;
+    for( const mutation_branch &mut : mutation_branch::get_all() ) {
+        if( !mut.threshold_substitutes.empty() ) {
+            if( has_trait( mut.id ) ) {
+                for( const trait_id &thresh_sub : mut.threshold_substitutes ) {
+                    if( has_trait( thresh_sub ) ) {
+                        ++count;
+                        break;
+                    }
+                }
+            }
+        }
+    }
+    return count;
```

Comment:
```suggestion
    int count = 0;
    for( const mutation_branch &mut : mutation_branch::get_all() ) {
        if( !mut.threshold_substitutes.empty() && has_trait( mut.id ) &&
            std::find_if( mut.threshold_substitutes.begin(), mut.threshold_substitutes.end(),
                          [this](const trait_id &t) { return has_trait( t ); } ) != mut.threshold_substitutes.end() ) {
            ++count;
            break;
        }
    }
    return count;
```
For a C++-y way to do it: (me has_trait is probably redundant, but I might be wrong)
```suggestion
    auto me_has_trait = [this](const trait_id &t) { return has_trait( t ); };
    auto pred = [this, me_has_trait](const mutation_branch &mut) {
        return has_trait( mut.id ) &&
               std::find_if( mut.threshold_substitutes.begin(), mut.threshold_substitutes.end(), me_has_trait ) != mut.threshold_substitutes.end();
    };
    return std::count_if( mutation_branch::get_all().begin(), mutation_branch::get_all().end(), pred );
```